akka-http/src/main/scala/akka/http/scaladsl/unmarshalling/PredefinedFromStringUnmarshallers.scala

```diff
@@ -45,14 +45,17 @@ trait PredefinedFromStringUnmarshallers {
       }
     }
 
-  implicit val uuidFromStringUnmarshaller: Unmarshaller[String, UUID] =
-    Unmarshaller.strict { string =>
-      try UUID.fromString(string)
-      catch {
-        case e: IllegalArgumentException =>
-          throw new IllegalArgumentException(s"'$string' is not a valid UUID value", e)
-      }
+  implicit val uuidFromStringUnmarshaller: Unmarshaller[String, UUID] = {
+    val validUuidRegex = "[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}"
+    val nilUuidString = "00000000-0000-0000-0000-000000000000"
+
+    Unmarshaller.strict {
+      case string if string.matches(validUuidRegex) | string == nilUuidString =>
```

Comment:
```suggestion
      case string if string == nilUuidString || string.matches(validUuidRegex) =>
```
Or since we're not actually pattern-matching, perhaps a 'regular' function might be fine here?

Reply:
done